src/sbd-md.c

```diff
@@ -1072,24 +1057,21 @@ int servant(const char *diskname, int mode, const void* argp)
 	/* FIXME: check error */
 	sigprocmask(SIG_SETMASK, &servant_masks, NULL);
 
-	atexit(servant_exit);
-	servant_inform_parent = 1;
-
 	st = open_device(diskname, LOG_WARNING);
 	if (!st) {
-		return -1;
+		exit(EXIT_MD_IO_FAIL);
```

Comment:
There's one return before this, if (!diskname). Shouldn't we exit there, too?

Also, with these changes, I think there's no distinction between exit() vs goto out -- probably stick with one or the other all the way through.